### PR TITLE
Add support for slash / hybrid commands

### PR DESCRIPTION
**Pull request by ArcasCZ**
Changes made so far:

1. flake8 and black first to make the checks happy (as it would probably fail)
2. updated discord.py to newest version (as we are gonna work with Slash commands)
3. added support for Interaction check into ACL2
4. fixed synchronization of slash / hybrid commands (on bot start + on module manipulation)
5. Converted ping command to hybrid.

As I'm working with ACL2 checks, even though I've tested it, I'd anyway recommend other people to test it out, just to be on the safe side.

### CODE REVIEW

**Comment by ArcasCZ** [comment]
To test this out, I recommend editing entrypoint.sh, line 35 to:

```
cd /pumpkin-py && trace_pie_acl=1 python3 pumpkin.py```

**Review by j-fiala — APPROVED** [review]
I've tested the updated discord.py version for Slash commands, ACL2 Interaction checks, and command synchronization fixes. Everything is functioning without errors. I approve the changes for merge.

File: pie/acl/__init__.py [hunk 1]
```diff
@@ -97,60 +97,85 @@ async def repeat(self, ctx, *, input: str):
...
+    bot: Union[commands.Bot, commands.AutoShardedBot],
+    invoker: Union[discord.User, discord.Member],
+    command: str,
+    guild: discord.Guild = None,
+    channel: discord.abc.Messageable = None,
 ) -> bool:
     """Check function based on Access Control.
 
     Set `for_command` to perform the check for other command
     then the one being invoked.
```
Comment (BorekZnovustvoritel) [inline pie/acl/__init__.py:142]: You have changed the function signature, but forgot to change the related docstring.

**Review by BorekZnovustvoritel — COMMENTED** [review]
The change makes sense overall and the ping command seems to work just alright. You have just forgot to update a docstring. Also I have a point to discussion:

Is there a reason to use `type(obj) is ...` instead of `isinstance(obj, ...)`? I know in the codebase we used the former, but I cannot really think of any reason to have it this way as this would break if Discord py were to subclass these data types.

File: pie/acl/__init__.py (hunk 1 above)
Comment (ArcasCZ) [reply pie/acl/__init__.py]: Updated.

**Pull request merged**